phoenix-core-client/src/main/java/org/apache/phoenix/execute/TupleProjector.java

```diff
@@ -322,6 +322,15 @@ public void getKey(ImmutableBytesWritable ptr) {
             ptr.set(keyPtr.get(), keyPtr.getOffset(), keyPtr.getLength());
         }
 
+        @Override
+        public long getKeyValueBytesSize() {
+            long size = keyPtr.getLength() + projectedValue.getLength();
+            if (keyValue != null) {
+                size += keyValue.getFamilyLength() + keyValue.getQualifierLength();
```

Comment:
Don't you need the length of the actual value object?

Reply:
For projection, we are taking projectedValue's length, which will cover actual value.